### PR TITLE
Increase usability

### PR DESCRIPTION
**Pull request by tgraf**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103852991%22%2C%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103854029%22%2C%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103854135%22%2C%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103925898%22%2C%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103926007%22%2C%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103926218%22%2C%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103936509%22%2C%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103936891%22%2C%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103937259%22%2C%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103964691%22%2C%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103964756%22%2C%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103965195%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%200beccd6fed51522b4e3c1360b74f023cca38d72a%20Godeps/Godeps.json%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103852991%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22You%20replaced%20the%20CLI%20lib%2C%20right%3F%20Why%20wasn%27t%20removed%20from%20this%20list%3F%22%2C%20%22created_at%22%3A%20%222017-03-02T05%3A18%3A24Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/5714066%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/aanm%22%7D%7D%2C%20%7B%22body%22%3A%20%22The%20docker%20plugin%20is%20still%20use%20urfave/cli%22%2C%20%22created_at%22%3A%20%222017-03-02T16%3A22%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/1417913%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tgraf%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Godeps/Godeps.json%3AL1-8%22%7D%2C%20%22Pull%20ebb3498725cfafd1ea774dd41325dba5fcf239ea%20daemon/main.go%20364%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103854135%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%60%5C%22Enables%20logstash%20agent%20for%20debug%20messages%5C%22%60%22%2C%20%22created_at%22%3A%20%222017-03-02T05%3A33%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/5714066%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/aanm%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20daemon/main.go%3AL12-152%22%7D%2C%20%22Pull%20ebb3498725cfafd1ea774dd41325dba5fcf239ea%20contrib/autocomplete/bash_autocomplete%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103926007%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%27ve%20fixed%20it%20up%20so%20it%20works%20for%20the%20new%20CLI%20lib%20as%20well.%22%2C%20%22created_at%22%3A%20%222017-03-02T13%3A44%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/1417913%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tgraf%22%7D%7D%2C%20%7B%22body%22%3A%20%22Changed%20this%20to%20generate%20the%20static%20bash%20completion%20once%20to%20not%20require%20to%20load%20the%20cilium%20binary%20for%20every%20attempted%20completion.%22%2C%20%22created_at%22%3A%20%222017-03-02T16%3A24%3A27Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/1417913%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tgraf%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20contrib/autocomplete/bash_autocomplete%3AL6-13%22%7D%2C%20%22Pull%20ebb3498725cfafd1ea774dd41325dba5fcf239ea%20contrib/upstart/cilium-net-daemon.conf%203%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103936891%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Fixed%22%2C%20%22created_at%22%3A%20%222017-03-02T14%3A36%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/1417913%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tgraf%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20contrib/upstart/cilium-net-daemon.conf%3AL1-3%22%7D%2C%20%22Pull%202b0d613c7208a7f912187c3eb7c8fe0c19fdc9e1%20contrib/scripts/install-llvm.sh%2026%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103964691%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Feel%20free%20to%20add%20it.%22%2C%20%22created_at%22%3A%20%222017-03-02T16%3A22%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/1417913%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tgraf%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20contrib/scripts/install-llvm.sh%3AL1-32%22%7D%2C%20%22Pull%20ebb3498725cfafd1ea774dd41325dba5fcf239ea%20cilium/cmd/helpers.go%2069%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103936509%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Fixed.%20I%27ve%20readded%20endpoint%20ID%20completion%20again.%22%2C%20%22created_at%22%3A%20%222017-03-02T14%3A34%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/1417913%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tgraf%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20cilium/cmd/helpers.go%3AL1-86%22%7D%2C%20%22Pull%20ebb3498725cfafd1ea774dd41325dba5fcf239ea%20daemon/main.go%20356%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103926218%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%27m%20fine%20with%20keeping%20it%20as%20%60k8s-api-server%60.%20I%20went%20back%20and%20forth%20on%20this%20one.%20I%27ll%20change%20it%20back.%22%2C%20%22created_at%22%3A%20%222017-03-02T13%3A45%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/1417913%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tgraf%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20daemon/main.go%3AL12-152%22%7D%2C%20%22Pull%20ebb3498725cfafd1ea774dd41325dba5fcf239ea%20daemon/main.go%20346%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103854029%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22should%20we%20get%20rid%20of%20all%20one%20letter%20flags%3F%22%2C%20%22created_at%22%3A%20%222017-03-02T05%3A31%3A39Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/5714066%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/aanm%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20kept%20those%20which%20are%20very%20frequently%20used.%20Debugging%2C%20device%2C%20...%22%2C%20%22created_at%22%3A%20%222017-03-02T13%3A44%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/1417913%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tgraf%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20daemon/main.go%3AL12-152%22%7D%2C%20%22Pull%20ebb3498725cfafd1ea774dd41325dba5fcf239ea%20daemon/options/options.go%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cilium/cilium/pull/294%23discussion_r103937259%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Those%20are%20files%20generated%20by%20the%20%60cobra%60%20binary.%20I%27ve%20fixed%20it%20up.%22%2C%20%22created_at%22%3A%20%222017-03-02T14%3A38%3A01Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/1417913%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tgraf%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20daemon/options/options.go%3AL1-46%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-Pull 0beccd6fed51522b4e3c1360b74f023cca38d72a Godeps/Godeps.json 1'></a> &nbsp;<b><a href='https://github.com/cilium/cilium/pull/294#discussion_r103852991'>File: Godeps/Godeps.json:L1-8</a></b>
- <a href='https://github.com/aanm'></a> You replaced the CLI lib, right? Why wasn't removed from this list?
- <a href='https://github.com/tgraf'></a> The docker plugin is still use urfave/cli
- [x] <a href='#crh-comment-Pull ebb3498725cfafd1ea774dd41325dba5fcf239ea daemon/main.go 346'></a> &nbsp;<b><a href='https://github.com/cilium/cilium/pull/294#discussion_r103854029'>File: daemon/main.go:L12-152</a></b>
- <a href='https://github.com/aanm'></a> should we get rid of all one letter flags?
- <a href='https://github.com/tgraf'></a> I kept those which are very frequently used. Debugging, device, ...
- [x] <a href='#crh-comment-Pull ebb3498725cfafd1ea774dd41325dba5fcf239ea daemon/main.go 364'></a> &nbsp;<b><a href='https://github.com/cilium/cilium/pull/294#discussion_r103854135'>File: daemon/main.go:L12-152</a></b>
- <a href='https://github.com/aanm'></a> `"Enables logstash agent for debug messages"`
- [x] <a href='#crh-comment-Pull ebb3498725cfafd1ea774dd41325dba5fcf239ea contrib/autocomplete/bash_autocomplete 1'></a> &nbsp;<b><a href='https://github.com/cilium/cilium/pull/294#discussion_r103926007'>File: contrib/autocomplete/bash_autocomplete:L6-13</a></b>
- <a href='https://github.com/tgraf'></a> I've fixed it up so it works for the new CLI lib as well.
- <a href='https://github.com/tgraf'></a> Changed this to generate the static bash completion once to not require to load the cilium binary for every attempted completion.
- [x] <a href='#crh-comment-Pull ebb3498725cfafd1ea774dd41325dba5fcf239ea daemon/main.go 356'></a> &nbsp;<b><a href='https://github.com/cilium/cilium/pull/294#discussion_r103926218'>File: daemon/main.go:L12-152</a></b>
- <a href='https://github.com/tgraf'></a> I'm fine with keeping it as `k8s-api-server`. I went back and forth on this one. I'll change it back.
- [x] <a href='#crh-comment-Pull ebb3498725cfafd1ea774dd41325dba5fcf239ea cilium/cmd/helpers.go 69'></a> &nbsp;<b><a href='https://github.com/cilium/cilium/pull/294#discussion_r103936509'>File: cilium/cmd/helpers.go:L1-86</a></b>
- <a href='https://github.com/tgraf'></a> Fixed. I've readded endpoint ID completion again.
- [x] <a href='#crh-comment-Pull ebb3498725cfafd1ea774dd41325dba5fcf239ea contrib/upstart/cilium-net-daemon.conf 3'></a> &nbsp;<b><a href='https://github.com/cilium/cilium/pull/294#discussion_r103936891'>File: contrib/upstart/cilium-net-daemon.conf:L1-3</a></b>
- <a href='https://github.com/tgraf'></a> Fixed
- [x] <a href='#crh-comment-Pull ebb3498725cfafd1ea774dd41325dba5fcf239ea daemon/options/options.go 1'></a> &nbsp;<b><a href='https://github.com/cilium/cilium/pull/294#discussion_r103937259'>File: daemon/options/options.go:L1-46</a></b>
- <a href='https://github.com/tgraf'></a> Those are files generated by the `cobra` binary. I've fixed it up.
- [x] <a href='#crh-comment-Pull 2b0d613c7208a7f912187c3eb7c8fe0c19fdc9e1 contrib/scripts/install-llvm.sh 26'></a> &nbsp;<b><a href='https://github.com/cilium/cilium/pull/294#discussion_r103964691'>File: contrib/scripts/install-llvm.sh:L1-32</a></b>
- <a href='https://github.com/tgraf'></a> Feel free to add it.

<a href='https://www.codereviewhub.com/cilium/cilium/pull/294?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/cilium/cilium/pull/294?approve=1'></a>&nbsp;<a href='https://github.com/cilium/cilium/pull/294'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by aanm** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/cilium/cilium/pull/294'>Refresh</a>.<a href='#crh-update-none'></a>

File: cilium/cmd/helpers.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	if len(args) < 1 {
+		Usagef(cmd, "Missing service id argument")
+	}
+
+	if args[0] == "" {
+		Usagef(cmd, "Empty service id argument")
+	}
+}
+
+//func listEndpointsBash() {
```
Comment (aanm) [inline cilium/cmd/helpers.go]: Missing TODO?

File: Godeps/Godeps.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
```
Comment (aanm) [inline Godeps/Godeps.json:1]: You replaced the CLI lib, right? Why wasn't removed from this list?

File: contrib/autocomplete/bash_autocomplete [hunk 3]
```diff
@@ -6,7 +6,7 @@ _cli_bash_autocomplete() {
      local cur opts base
```
Comment (aanm) [inline contrib/autocomplete/bash_autocomplete]: This file was for the old CLI lib so I think you can remove it.

https://github.com/urfave/cli#distribution

File: contrib/upstart/cilium-net-daemon.conf [hunk 4]
```diff
@@ -1,2 +1,2 @@
 env PATH=/usr/local/clang/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin
-exec cilium -D daemon run --consul-server "127.0.0.1:8500"
+exec cilium-agent ---debug --consul "127.0.0.1:8500"
```
Comment (aanm) [inline contrib/upstart/cilium-net-daemon.conf]: triple `-`

File: daemon/main.go [hunk 5]
```diff
@@ -12,362 +12,140 @@
...
+	flags := RootCmd.Flags()
+	flags.StringVar(&cfgFile, "config", "", "config file (default is $HOME/ciliumd.yaml)")
+	flags.BoolP("debug", "D", false, "Enable debug messages")
+	flags.StringVar(&consulAddr, "consul", "", "Consul agent address [127.0.0.1:8500]")
+	flags.StringVarP(&config.Device, "device", "d", "undefined", "Device to snoop on")
+	flags.BoolVar(&disableConntrack, "disable-conntrack", false, "Disable connection tracking")
+	flags.BoolVar(&disablePolicy, "disable-policy", false, "Disable policy enforcement")
+	flags.StringVarP(&config.DockerEndpoint, "docker", "e", "unix:///var/run/docker.sock",
+		"Register a listener for docker events on the given endpoint")
+	flags.StringSliceVar(&etcdAddr, "etcd", []string{}, "Etcd agent address [http://127.0.0.1:2379]")
```
Comment (aanm) [inline daemon/main.go]: `etcd-server` maybe?

File: daemon/main.go [hunk 6]
```diff
@@ -12,362 +12,140 @@
...
+func init() {
+	cobra.OnInitialize(initConfig)
+	flags := RootCmd.Flags()
+	flags.StringVar(&cfgFile, "config", "", "config file (default is $HOME/ciliumd.yaml)")
+	flags.BoolP("debug", "D", false, "Enable debug messages")
+	flags.StringVar(&consulAddr, "consul", "", "Consul agent address [127.0.0.1:8500]")
+	flags.StringVarP(&config.Device, "device", "d", "undefined", "Device to snoop on")
+	flags.BoolVar(&disableConntrack, "disable-conntrack", false, "Disable connection tracking")
+	flags.BoolVar(&disablePolicy, "disable-policy", false, "Disable policy enforcement")
+	flags.StringVarP(&config.DockerEndpoint, "docker", "e", "unix:///var/run/docker.sock",
```
Comment (aanm) [inline daemon/main.go]: can you remove the `e`?

File: daemon/main.go [hunk 7]
```diff
@@ -12,362 +12,140 @@
...
-
-		fmt.Printf("%-24s %s\n", k, text)
-	}
+func init() {
+	cobra.OnInitialize(initConfig)
+	flags := RootCmd.Flags()
+	flags.StringVar(&cfgFile, "config", "", "config file (default is $HOME/ciliumd.yaml)")
+	flags.BoolP("debug", "D", false, "Enable debug messages")
+	flags.StringVar(&consulAddr, "consul", "", "Consul agent address [127.0.0.1:8500]")
+	flags.StringVarP(&config.Device, "device", "d", "undefined", "Device to snoop on")
```
Comment (aanm) [inline daemon/main.go:91]: should we get rid of all one letter flags?

File: daemon/main.go [hunk 8]
```diff
@@ -12,362 +12,140 @@
...
+	flags.BoolVar(&disableConntrack, "disable-conntrack", false, "Disable connection tracking")
+	flags.BoolVar(&disablePolicy, "disable-policy", false, "Disable policy enforcement")
+	flags.StringVarP(&config.DockerEndpoint, "docker", "e", "unix:///var/run/docker.sock",
+		"Register a listener for docker events on the given endpoint")
+	flags.StringSliceVar(&etcdAddr, "etcd", []string{}, "Etcd agent address [http://127.0.0.1:2379]")
+	flags.StringVar(&config.EtcdCfgPath, "etcd-config-path", "", "Absolute path to the etcd configuration file")
+	flags.BoolVar(&enableTracing, "enable-tracing", false, "Enable tracing while determining policy")
+	flags.StringVar(&nat46prefix, "nat46-range", addressing.DefaultNAT46Prefix,
+		"IPv6 prefix to map IPv4 addresses to")
+	flags.StringVar(&config.K8sEndpoint, "k8s-api", "", "Kubernetes api address server")
```
Comment (aanm) [inline daemon/main.go]: `k8s-api-server` maybe?

File: daemon/main.go [hunk 9]
```diff
@@ -12,362 +12,140 @@
...
+	flags.StringVar(&config.K8sCfgPath, "k8s-kubeconfig-path", "", "Absolute path to the kubeconfig file")
+	flags.StringSliceVar(&k8sLabels, "k8s-prefix", []string{},
+		"Key values that will be read from kubernetes. (Default: k8s-app, version)")
+	flags.BoolVar(&config.KeepConfig, "keep-config", false,
+		"When restoring state, keeps containers' configuration in place")
+	flags.StringVar(&labelPrefixFile, "label-prefix-file", "", "File with valid label prefixes")
+	flags.StringVar(&config.LibDir, "libdir", defaults.LibDir, "Path to directory with program templates")
+	flags.BoolVar(&enableLogstash, "logstash", false, "Enables logstash agent")
```
Comment (aanm) [inline daemon/main.go]: `"Enables logstash agent for debug messages"`

File: daemon/main.go [hunk 10]
```diff
@@ -12,362 +12,140 @@
...
+	flags.StringVar(&logstashAddr, "logstash-agent", "127.0.0.1:8080", "Logstash agent address")
+	flags.Uint32Var(&logstashProbeTimer, "logstash-probe-timer", 10, "Logstash probe timer (seconds)")
+	flags.StringVarP(&v6Address, "node-address", "n", "", "IPv6 address of node, must be in correct format")
```
Comment (aanm) [inline daemon/main.go]: Remove `n`?

File: daemon/main.go [hunk 11]
```diff
@@ -12,362 +12,140 @@
...
+	flags.BoolVar(&config.RestoreState, "restore", false,
+		"Restores state, if possible, from previous daemon")
+	flags.StringVar(&config.RunDir, "state-dir", defaults.RuntimePath, "Path to directory to store runtime state")
+	flags.StringVar(&socketPath, "socket-path", defaults.SockPath, "Sets the socket path to listen for connections")
+	flags.BoolVar(&config.LBMode, "lb", false,
+		"Enables load balancer mode where load balancer bpf program is attached to the interface ")
+	flags.BoolVar(&config.IPv4Disabled, "disable-ipv4", false, "Disable IPv4 mode")
+	flags.StringVar(&v4Prefix, "ipv4-range", "", "IPv4 prefix")
+	flags.StringVarP(&config.Tunnel, "tunnel", "t", "vxlan", "Tunnel mode vxlan or geneve, vxlan is the default")
```
Comment (aanm) [inline daemon/main.go:121]: Remove `t`?

File: daemon/options/options.go [hunk 12]
```diff
@@ -0,0 +1,45 @@
+// Copyright 2016-2017 Authors of Cilium
```
Comment (aanm) [inline daemon/options/options.go:1]: There are some files where you have `Copyright (c) ...` and this one have `Copyright 2016-...`

File: pkg/kvstore/local.go [hunk 13]
```diff
@@ -0,0 +1,262 @@
+// Copyright 2016-2017 Authors of Cilium
```
Comment (aanm) [inline pkg/kvstore/local.go:1]: `Copyright (c)` thing..

File: contrib/scripts/install-llvm.sh [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+CLANG_URL="http://releases.llvm.org/3.8.1/$CLANG_FILE"
+CLANGROOT=/usr/local/clang
+
+if [ -d "/usr/local/$CLANG_DIR" ]; then
+	echo "Existing clang installation found."
+	echo "Removing old installation..."
+	rm -rf /usr/local/$CLANG_DIR
+fi
+
+wget $CLANG_URL
```
Comment (aanm) [inline contrib/scripts/install-llvm.sh:26]: Since the url is not https, should we check the checksum file?

**Review by aanm — CHANGES_REQUESTED** [review]

File: daemon/main.go (hunk 7 above)
Comment (tgraf) [reply daemon/main.go:91]: I kept those which are very frequently used. Debugging, device, ...

File: contrib/autocomplete/bash_autocomplete (hunk 3 above)
Comment (tgraf) [reply contrib/autocomplete/bash_autocomplete]: I've fixed it up so it works for the new CLI lib as well.

File: daemon/main.go (hunk 8 above)
Comment (tgraf) [reply daemon/main.go]: I'm fine with keeping it as `k8s-api-server`. I went back and forth on this one. I'll change it back.

File: cilium/cmd/helpers.go (hunk 1 above)
Comment (tgraf) [reply cilium/cmd/helpers.go]: Fixed. I've readded endpoint ID completion again.

File: contrib/upstart/cilium-net-daemon.conf (hunk 4 above)
Comment (tgraf) [reply contrib/upstart/cilium-net-daemon.conf]: Fixed

File: daemon/options/options.go (hunk 12 above)
Comment (tgraf) [reply daemon/options/options.go:1]: Those are files generated by the `cobra` binary. I've fixed it up.

File: contrib/scripts/install-llvm.sh (hunk 14 above)
Comment (tgraf) [reply contrib/scripts/install-llvm.sh:26]: Feel free to add it.

File: Godeps/Godeps.json (hunk 2 above)
Comment (tgraf) [reply Godeps/Godeps.json:1]: The docker plugin is still use urfave/cli

File: contrib/autocomplete/bash_autocomplete (hunk 3 above)
Comment (tgraf) [reply contrib/autocomplete/bash_autocomplete]: Changed this to generate the static bash completion once to not require to load the cilium binary for every attempted completion.

**Review by aanm — APPROVED** [review]

**Pull request merged**